lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1146,6 +1146,21 @@ sub create_bam_header {
     return 1;
 }
 
+sub set_bam_size {
+    my ($self, $bam_file) = @_;
+    return 1 if $self->bam_size;
+
+    unless (defined $bam_file && -s $bam_file) {
+        $bam_file = $self->get_bam_file;
+        unless (-s $bam_file) {
+            die $self->error_message('BAM file (%s) does not exist or is empty', $bam_file);
+        }
+    }
+
```

Comment:
A lock is probably needed here. Also check bam_size again since revivifying bam of get_bam_file could take long time and bam_size already set
my $lock  = $self->get_bam_lock;
my $guard = Scope::Guard->new(sub { $lock->unlock() });
return 1 if $self->bam_size;